$DragonFly: src/secure/usr.sbin/sshd/Attic/sshd.c.patch,v 1.1 2004/07/31 20:05:00 geekgod Exp $
--- sshd.c.orig	2004-07-23 14:26:51.000000000 +0200
+++ sshd.c	2004-07-23 14:20:00.000000000 +0200
@@ -53,6 +53,10 @@
 #include <prot.h>
 #endif
 
+#ifdef __DragonFly__
+#include <resolv.h>
+#endif
+
 #include "ssh.h"
 #include "ssh1.h"
 #include "ssh2.h"
@@ -1415,6 +1419,17 @@
 	    sizeof(on)) < 0)
 		error("setsockopt SO_KEEPALIVE: %.100s", strerror(errno));
 
+#ifdef __DragonFly__
+	/*
+	 * Initialize the resolver.  This may not happen automatically
+	 * before privsep chroot().                                   
+	 */
+	if ((_res.options & RES_INIT) == 0) {
+		debug("res_init()");         
+		res_init();         
+	}
+#endif
+
 	/*
 	 * Register our connection.  This turns encryption off because we do
 	 * not have a key.
